index.d.ts

```diff
@@ -1130,7 +1130,7 @@ export interface ExistsFunction<
 
 export interface i18n {
   // Expose parameterized t in the i18next interface hierarchy
-  t: TFunction<FallbackOrNS<string>[]>;
+  t: TFunction<Namespace | DefaultNamespace>;
```

Comment:
Hey @daveallie, thank you for your contribution!
Unfortunately, the approach here won't work as expected. `Namespace` will return all namespaces, which means that all namespaces will be considered "Default Namespaces".
Additionally, we should filter the `DefaultNamespace` out if the `DefaultNamespace` is not included in the resources. Otherwise, it'll infer the namespace as `string`, which means that `t` function will accept any key.
Both problems can be easily solved. My biggest concern is regarding the compilation time since we'll iterate over the default translation keys twice. It can be solved but may require major changes in the Normalize type.